client/internal/clientcommon.go

```diff
@@ -97,12 +98,17 @@ func (c *ClientCommon) PrepareStart(
 	if c.isStarted {
 		return errAlreadyStarted
 	}
-	capabilities := settings.Capabilities
-
-	// According to OpAMP spec this capability MUST be set, since all Agents MUST report status.
-	capabilities |= protobufs.AgentCapabilities_AgentCapabilities_ReportsStatus
-	if err := c.ClientSyncedState.SetCapabilities(&capabilities); err != nil {
-		return err
+	// Deprecated: Use client.SetCapabilities() instead.
+	if settings.Capabilities != 0 {
```

Comment:
@tigrannajaryan this _technically_ could be a breaking change if someone isn't setting any capabilities, should I just default to setting reports status here and then in the future we remove this?

Reply:
I think we should emit a warning in the log if SetCapabilities is not called before Start. Later, we can make it an error. Then finally some time after that we remove Settings.Capabilities field.

In any of these 3 states, ReportsStatus should always be automatically added, regardless of how the Capabilities are set (via Settings or via method call).

Reply:
@tigrannajaryan i do make sure reportsstatus is always set (I do the add in this method as well in the clientcommon setcapabilities), but I'll add in a log right now.

Reply:
@tigrannajaryan should be all good now. PTAL 🙇